src/core/3d/qgs3dviewsmanager.h

```diff
@@ -0,0 +1,101 @@
+/***************************************************************************
+    qgs3dviewsmanager.h
+    ------------------
+    Date                 : December 2021
+    Copyright            : (C) 2021 Belgacem Nedjima
+    Email                : gb underscore nedjima at esi dot dz
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGS3DVIEWSMANAGER_H
+#define QGS3DVIEWSMANAGER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include <QObject>
+#include <QDomDocument>
+#include <QDomElement>
+#include <QMap>
+
+class QgsProject;
+
+/**
+ * \ingroup core
+ * \class Qgs3DViewsManager
+ *
+ * \brief Manages storage of a set of 3D views.
+ *
+ * QgsLayoutManager handles the storage, serializing and deserializing
+ * of 3D views. Usually this class is not constructed directly, but
+ * rather accessed through a QgsProject via QgsProject::viewsManager3D().
+ *
+ * \since QGIS 3.24
+ */
+class CORE_EXPORT Qgs3DViewsManager : public QObject
```

Comment:
Can we rename this as QgsMapViewManager, so that in future we can extend to include storage of 2d views as well? (also move to src/core/project accordingly)